### PR TITLE
fix: Add Date and Moment as valid types for chartjs-plugin-annotation…

### LINKED ISSUE #47274: Missing types
- [x] I tried using the `@types/chartjs-plugin-annotation` package and had problems.
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @danmana 
  - PR: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/44114

`BoxAnnotationOptions.xMin` and `xMax` don't accept `Date` nor `Moment` types.

### PR DESCRIPTION
**Pull request by danmana**
… xMin, xMax. Closes #47274

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/chartjs/chartjs-plugin-annotation/blob/v0.5.7/samples/line-time-scale.html#L171
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)

### CODE REVIEW

File: types/chartjs-plugin-annotation/tslint.json [hunk 1]
```diff
@@ -1 +1,6 @@
-{ "extends": "dtslint/dt.json" }
+{
+  "extends": "dtslint/dt.json",
+  "rules": {
+    "no-outside-dependencies": false
+  }
+}
```
Comment (peterblazejewicz) [inline types/chartjs-plugin-annotation/tslint.json:6]:
```suggestion
{
  "extends": "dtslint/dt.json"
}
```
what about defaulting to 'default' settings? 
It obviously would work on CI as dependencies are being installed for CI builds,
on the local machine it's enough to run `npm i --no-package-lock` for each depenedency, here for `moment' in `types/chartjs-plugin-annotation` and for 'chart.js` in `types/chart.js`.
Could using this exclusion prevent us from catching future errors?

**Review by peterblazejewicz — COMMENTED** [review]
@danmana LGTM, just this linting exclusion issue, mind reviewing comment?

File: types/chartjs-plugin-annotation/tslint.json (hunk 1 above)
Comment (danmana) [reply types/chartjs-plugin-annotation/tslint.json:6]: hi @peterblazejewicz 

I'm not really familiar with how the CI build works with type dependencies that are defined outside DefinetlyTyped, I just copied what I saw other type definitions doing when working with `moment`.

I looked at a few, including `chart.js` and all seemed to have this combination: `moment` in package.json, `no-outside-dependencies` in tslint.json.

I agree, if this works without setting it, then we should keep tslint.json minimal.

Let's try it and see if it works :)


PS: do you know if there are some docs on how we should work with external type dependencies? and when does [allowedPackageJsonDependencies.txt](https://raw.githubusercontent.com/microsoft/DefinitelyTyped-tools/master/packages/definitions-parser/allowedPackageJsonDependencies.txt) come into play?

**Review by peterblazejewicz — APPROVED** [review]
LGTM!

**Pull request merged**